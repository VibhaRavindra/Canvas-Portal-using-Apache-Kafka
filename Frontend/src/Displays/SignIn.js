import React, { Component } from 'react';
import '../App.css';
import { Redirect, Link } from 'react-router-dom'
import { signIn } from "../js/actions/actions";
import { connect } from "react-redux";

function mapStateToProps(state) {
    return {isLoggedIn:state.isLoggedIn,
        userid:state.userid,
        IsInvalid:state.IsInvalid,
    errMessage:state.errMessage}
}

function mapDispatchToProps(dispatch) {
    return {
        signIn: (data) => dispatch(signIn(data))
    };
}
class SignIn extends Component {
    constructor(props){
        super(props);
        this.submitSignIn = this.submitSignIn.bind(this);
    }
    async submitSignIn(e){
        e.preventDefault();
        const data = new FormData(e.target);
        this.props.signIn(data);
    }
    
    render(){
        console.log(this.props)
        let redirectRender, redirectDashboard, failCase;
        if(this.props.isLoggedIn === true && this.props.IsInvalid === false){
            redirectDashboard = <Redirect to = '/dashboard' />
        }else if (this.props.IsInvalid === true){
            failCase = <div className="error-box">
            <span>
            <img className="error-icon" src="https://www.freeiconspng.com/uploads/alert-icon-red-11.png" alt="" />
            </span>
            <span className="error-msg">{this.props.errMessage}</span>
        </div>
        }
        
        return(
            <div>
                {redirectDashboard}
            <div className="LoginPage">
                <div className="imageHeader">
                    <img className="sjsuImage" src="https://ok2static.oktacdn.com/bc/image/fileStoreRecord?id=fs01heub3azJBMXWF0x7" alt="" />
                    <div className="beacon-container"></div>
                </div>
                
                <div className="LogIncontainer">
                    <p className="signInText">Sign In</p>
                    {failCase}
                    <form className="form" onSubmit={this.submitSignIn}>
                    <input type="email" className="inputStyle1" name="emailId" placeholder="SJSU Email ID" required />                
                    <input type="password" className="inputStyle1" name="password" placeholder="Password" required /> 
                    <button type="submit" className="signInBtn">Sign In</button>
                    <Link to="/signup"><button className="orText" >Create an account? </button></Link>
                    {redirectRender}
                    </form>
                </div> 
            </div>
            </div>
        )
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(SignIn);