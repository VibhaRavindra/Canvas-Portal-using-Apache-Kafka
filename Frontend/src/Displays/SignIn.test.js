import React from 'react';
import SignIn from './SignIn';
import renderer from 'react-test-renderer';
import { mount, shallow, render } from 'enzyme';
import { Provider } from "react-redux";
import store from "../js/store/store";
import { BrowserRouter as Router } from "react-router-dom";
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

it('Sign in renders correctly', () => {
    const tree = renderer
      .create(<Provider store={store}><Router><SignIn/></Router></Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

//   it('Sign in renders correctly', () => {
//     const wrapper = mount(<SignIn/>);
//     wrapper.find('#signupbutton').simulate('click');
//     expect(wrapper).toMatchSnapshot();
//   });