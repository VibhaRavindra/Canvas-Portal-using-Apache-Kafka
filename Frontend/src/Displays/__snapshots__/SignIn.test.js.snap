// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign in renders correctly 1`] = `
<div>
  <div
    className="LoginPage"
  >
    <div
      className="imageHeader"
    >
      <img
        alt=""
        className="sjsuImage"
        src="https://ok2static.oktacdn.com/bc/image/fileStoreRecord?id=fs01heub3azJBMXWF0x7"
      />
      <div
        className="beacon-container"
      />
    </div>
    <div
      className="LogIncontainer"
    >
      <p
        className="signInText"
      >
        Sign In
      </p>
      <form
        className="form"
        onSubmit={[Function]}
      >
        <input
          className="inputStyle1"
          name="emailId"
          placeholder="SJSU Email ID"
          required={true}
          type="email"
        />
        <input
          className="inputStyle1"
          name="password"
          placeholder="Password"
          required={true}
          type="password"
        />
        <button
          className="signInBtn"
          type="submit"
        >
          Sign In
        </button>
        <a
          href="/signup"
          onClick={[Function]}
        >
          <button
            className="orText"
          >
            Create an account? 
          </button>
        </a>
      </form>
    </div>
  </div>
</div>
`;
